tasks/safe_restart.rb

```diff
@@ -17,14 +17,13 @@
     ecs_gateway = Gateway::Aws::Ecs.new(config: { region: region })
     route53_gateway = Gateway::Aws::Route53.new
     cluster_finder = UseCase::FindClustersForEnvironment.new(gateway: ecs_gateway, environment: environment)
-    health_checker = UseCase::HealthCheck.new(route53_gateway: route53_gateway)
-    delayer = Gateway::Delayer.new
+    health_checker = UseCase::HealthCheck.new(route53_gateway: route53_gateway, delayer: Gateway::Delayer.new)
 
     UseCase::SafeRestart.new(
       cluster_finder: cluster_finder,
       ecs_gateway: ecs_gateway,
       health_checker: health_checker,
-      delayer: delayer,
+      delayer: Gateway::Delayer.new,
```

Comment:
Do we no longer need the `wait_time: 5` option for `Gateway::Delayer.new`?

Reply:
Yeah this object isn't constructed with a wait time, it is executed with an argument to the .delay method, setting the wait time.

As a rule of thumb, initializers are just for composing dependencies.  Executing the object with different arguments goes in the public method.  (I didn't do this previously)